swift_package/BrainFlow/Sources/BrainFlowBindings/DataFilter.swift

```diff
@@ -0,0 +1,650 @@
+    }
+
+    /**
+     * set log level
+     */
+    static func setLogLevel (_ logLevel: LogLevels) throws {
+        let errorCode = set_log_level_data_handler (logLevel.rawValue)
+        try checkErrorCode("Error in set_log_level", errorCode)
+    }
+
+    /**
+    * perform lowpass filter in-place
+    */
+    static func performLowpass (data: inout [Double], samplingRate: Int32, cutoff: Double,
+                         order: Int32, filterType: FilterTypes, ripple: Double) throws {
+        let dataLen = Int32(data.count)
+        let filterVal = filterType.rawValue
+        let errorCode = perform_lowpass (&data, dataLen, samplingRate, cutoff, order, filterVal, ripple)
+        try checkErrorCode("Failed to apply filter", errorCode)
+    }
+
+    /**
+    * perform highpass filter in-place
+    */
+    static func performHighpass (data: inout [Double], samplingRate: Int32, cutoff: Double,
+                         order: Int32, filterType: FilterTypes, ripple: Double) throws {
+        let dataLen = Int32(data.count)
+        let filterVal = filterType.rawValue
+        let errorCode = perform_highpass (&data, dataLen, samplingRate, cutoff, order, filterVal, ripple)
+        try checkErrorCode("Failed to apply filter", errorCode)
+    }
+
+    /**
+     * perform bandpass filter in-place
+     */
+    static func performBandpass (data: inout [Double], samplingRate: Int32, startFreq: Double, stopFreq: Double,
+                          order: Int32, filterType: FilterTypes, ripple: Double) throws {
+        let dataLen = Int32(data.count)
+        let filterVal = filterType.rawValue
+        let errorCode = perform_bandpass(&data, dataLen, samplingRate, startFreq,
+                                         stopFreq, order, filterVal, ripple)
+        try checkErrorCode("Failed to apply filter", errorCode)
+    }
+
+    /**
+     * perform bandstop filter in-place
+     */
+    static func performBandstop (data: inout [Double], samplingRate: Int32, startFreq: Double, stopFreq: Double,
+                          order: Int32, filterType: FilterTypes, ripple: Double) throws {
+        let dataLen = Int32(data.count)
+        let filterVal = filterType.rawValue
+        let errorCode = perform_bandstop(&data, dataLen, samplingRate, startFreq,
+                                         stopFreq, order, filterVal, ripple)
+        try checkErrorCode("Failed to apply filter", errorCode)
+    }
+    
+    /**
+     * perform moving average or moving median filter in-place
+     */
+    static func performRollingFilter (data: inout [Double], period: Int32, operation: AggOperations) throws {
+        let dataLen = Int32(data.count)
+        let errorCode = perform_rolling_filter (&data, dataLen, period, operation.rawValue)
+        try checkErrorCode("Failed to apply filter", errorCode)
+    }
+    
+    /**
+     * subtract trend from data in-place
+     */
+    static func deTrend (data: inout [Double], operation: DetrendOperations) throws {
```

Comment:
detrend, all lowercase

Reply:
This one will be difficult. It won't compile due to a conflict with the C++ function of the same name.